cache.go

```diff
@@ -294,6 +294,37 @@ func (c *Cache[K, V]) Delete(key K) {
 	c.evict(EvictionReasonDeleted, elem)
 }
 
+// GetOrSet returns the existing value for the key if present.
+// Otherwise, it sets and returns the given value. The retrieved
+// result is true if the value was retrieved, false if set.
+func (c *Cache[K, V]) GetOrSet(key K, value V, ttl time.Duration) (*Item[K, V], bool) {
```

Comment:
`Get()` options (like on line 252) should probably be accepted as well. I just wonder what would be the best approach to do that. Maybe we should have `getWithOpts` that would contain the same logic as `Get` and call it in `Get`, `GetOrSet` and `GetAndDelete`?

Reply:
yeah that makes sense! updated. One small issue I can see is that there is a gap in the locking for `GetOrSet` between getting and setting which might not matter anyway?

Reply:
This might still cause unexpected problems. I think we can merge this PR without proper locking, but we should figure out how to update this before creating a new release.